slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -319,28 +329,47 @@ bool SLAMServiceImpl::ExtractPointCloudToBuffer(std::stringbuf &buffer) {
 void SLAMServiceImpl::ProcessData() {
     // Set up and build the MapBuilder
     SetUpMapBuilder();
+    DetermineActionMode();
+
+    if (action_mode == SLAMServiceActionMode::UPDATING ||
```

Comment:
[nit] use GetActionMode or remove function if unused

Reply:
`GetActionMode` is used in a few places outside of `slam_service.cc` (in tests and in `config.cc`). I think since this is implemented within the slam service, we can use the member variable directly instead of calling the function and saving the variable in a new variable name. I went through slam_service to make sure we're not using `GetActionMode` within its functions (doing that was an artifact from how it was implemented before).

Reply:
Oh darn, I github's search was not returning any results for it so I checked in main.cc and map_builder.cc and didnt see it being used. Didnt check config